src/main/java/com/bank/project/bank/dao/UserDao.java

```diff
@@ -0,0 +1,10 @@
+package com.bank.project.bank.dao;
+
+import com.bank.project.bank.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import java.util.Optional;
+
+public interface UserDao extends JpaRepository<User, Long> {
+
+    Optional<User> findByPhoneNumber(String phoneNumber);
```

Comment:
see checklist - in this method and also in `getById()` - we should join fetch roles as well

Reply:
Ok